src/components/command-bar/commands/search/unified-search/components/dialog-body/helpers/use-command-bar-product-tag.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useMemo } from 'react'
+import { useCommandBar } from 'components/command-bar'
+import { useCurrentProduct } from 'contexts'
+import { getCurrentProductTag } from '../../../../helpers'
+import { useSetUpAndCleanUpCommandState } from 'components/command-bar/hooks'
+
+/**
+ * Set and un-set the command-bar-level product filtering tag.
+ *
+ * TODO: consider refactoring this.
+ *
+ * We've had feedback from design at least that the "product tag" interface
+ * doesn't feel right within the search modal. This raises a few questions:
+ *
+ * - Is this genuinely a command-bar-wide setting, or are "product tag" filters
+ *   something we could apply to the "Search" command only for now? (Note:
+ *   I'm not even sure other "commands" (ie "settings") are implemented yet...
+ *   managing the "tags" at the command bar level feels like it may have
+ *   been a premature abstraction based on relatively speculative use cases).
+ *
+ * - In the context of search, specifically auto-filtering to a given product
+ *   when in that product context, Would typing-based filters, such as
+ *   `product:<productSlug>`, be preferable to the "product tag" approach?
+ */
+export function useCommandBarProductTag() {
+	const currentProduct = useCurrentProduct()
+	const { addTag, currentTags, removeTag } = useCommandBar()
+
+	/**
+	 * Create callback for setting up this command's state.
+	 */
+	const setUpCommandState = useCallback(() => {
+		if (currentProduct) {
+			addTag({
+				id: currentProduct.slug,
+				text: currentProduct.slug === 'hcp' ? 'HCP' : currentProduct.name,
+			})
+		}
+	}, [addTag, currentProduct])
+
+	/**
+	 * Create callback for cleaning up this command's state.
+	 */
+	const cleanUpCommandState = useCallback(() => {
+		if (currentProduct) {
+			removeTag(currentProduct.slug)
+		}
+	}, [currentProduct, removeTag])
+
+	/**
+	 * Leveraging the set up + clean up hook exposed by CommandBarDialog.
+	 */
+	useSetUpAndCleanUpCommandState(setUpCommandState, cleanUpCommandState)
+
+	/**
+	 * Get the CommandBarTag object for the current product if it's present.
+	 */
+	const currentProductTag = useMemo(
+		() =>
+			getCurrentProductTag({
+				currentProduct,
+				currentTags,
+			}),
+		[currentProduct, currentTags]
+	)
+
+	return currentProductTag
```

Comment:
💭 do we plan to unify this logic later on with the _non-unified-search_ `dialog-body`? [Perhaps a breadcrumb ](https://github.com/hashicorp/dev-portal/blob/5d49306ec4a582efa590f1fee0f81030cfb08c01/src/components/command-bar/commands/search/components/dialog-body/index.tsx#L161-L210)to the other file would be good.

Reply:
Or is the plan to fully delete the old non-unified search once this is ready?

Reply:
Plan is the latter, delete the non-unified search! ☺️